hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/InputPathHandler.java

```diff
@@ -117,9 +124,11 @@ private void parseInputPaths(Path[] inputPaths, List<String> incrementalTables)
     }
   }
 
-  private void tagAsIncrementalOrSnapshot(Path inputPath, String tableName,
-      HoodieTableMetaClient metaClient, List<String> incrementalTables) {
-    if (!incrementalTables.contains(tableName)) {
+  private void tagAsIncrementalOrSnapshot(Path inputPath, HoodieTableMetaClient metaClient, List<String> incrementalTables) {
+    String databaseName = metaClient.getTableConfig().getDatabaseName();
+    String tableName = metaClient.getTableConfig().getTableName();
+    if ((isIncrementalUseDatabase && !StringUtils.isNullOrEmpty(databaseName) && !incrementalTables.contains(databaseName + "." + tableName))
+            || (!(isIncrementalUseDatabase && !StringUtils.isNullOrEmpty(databaseName)) && !incrementalTables.contains(tableName))) {
```

Comment:
this condition check is pretty hard to read.. can we improve this?

Reply:
It has been modified to improve through the helper method